src/ts/components/seekbarcontrols.ts

```diff
@@ -0,0 +1,108 @@
+import { UIUtils } from '../uiutils';
+import { DOM } from '../dom';
+import { PlayerAPI } from 'bitmovin-player';
+import { VolumeController } from '../volumecontroller';
+
+export enum SeekBarType {
+    Vod,
+    Live,
+    Volume,
+}
+
+const changeRangeValue = (
+    value: number,
+    range: { min: number, max: number },
+    cb: (value: number) => void,
+) => {
+    if (value < range.min) {
+        cb(range.min);
+    } else if (value > range.max) {
+        cb(range.max);
+    } else {
+        cb(value);
+    }
+};
+
+const arrowKeyControls = (
+    currentValue: number,
+    range: { min: number, max: number },
+    keyStepIncrements: { leftRight: number, upDown: number },
+    valueUpdate: (value: number) => void,
+) => {
+    const controlValue = Math.floor(currentValue);
+
+    return {
+        left: () => changeRangeValue(controlValue - keyStepIncrements.leftRight, range, valueUpdate),
+        right: () => changeRangeValue(controlValue + keyStepIncrements.leftRight, range, valueUpdate),
+        up: () => changeRangeValue(controlValue + keyStepIncrements.upDown, range, valueUpdate),
+        down: () => changeRangeValue(controlValue - keyStepIncrements.upDown, range, valueUpdate),
+        home: () => changeRangeValue(range.min, range, valueUpdate),
+        end: () => changeRangeValue(range.max, range, valueUpdate),
+    };
+};
+
+const seekBarControls = (
+    type: SeekBarType,
+    keyStepIncrements: { leftRight: number, upDown: number },
+    player: PlayerAPI,
+    volumeController: VolumeController,
+) => {
+    if (type === SeekBarType.Live) {
+        return arrowKeyControls(player.getTimeShift(), { min: player.getMaxTimeShift(), max: 0 }, keyStepIncrements, player.timeShift);
+    } else if (type === SeekBarType.Vod) {
+        return arrowKeyControls(player.getCurrentTime(), { min: 0, max: player.getDuration() }, keyStepIncrements, player.seek);
+    } else if (type === SeekBarType.Volume && volumeController != null) {
+        const volumeTransition = volumeController.startTransition();
+        return arrowKeyControls(player.getVolume(), { min: 0, max: 100 }, keyStepIncrements, volumeTransition.finish.bind(volumeTransition));
+    }
+};
+
+export const setSeekBarControls = (
```

Comment:
Why is this specifically attached to the seek bar? I think this should be a separate controller, e.g. attached to the `UIContainer`, that issues commands directly on the player. The `SeekBar`, like any other `Component` in this UI framework, will then simply listen to and react upon the player events.

Reply:
We discussed this with @dweinber, we decided that the UI wouldn't have controls on the whole player because our customers had different requirements and specifications for it. That is why we agreed on adding controls on on the sliders

Reply:
Then this should be configurable. Needing to focus the seekbar to issue unrelated actions is bad UX.